applications/OptimizationApplication/python_scripts/controls/thickness/shell_thickness_control.py

```diff
@@ -0,0 +1,235 @@
+     {
+        "solver_settings" : {
+             "domain_size"          : 3,
+             "echo_level"           : 0,
+             "filter_type"          : "general_scalar",
+             "model_import_settings": {
+                 "input_type"     : "use_input_model_part"
+             }
+         },
+        "problem_data": {
+             "echo_level"    : 0,
+             "start_time"    : 0.0,
+             "end_time"      : 1.0,
+             "parallel_type" : "OpenMP"
+         },
+        "processes" : {
+                "boundary_conditions_process_list" : []
+            }
+     }""")
+
+    shell_scalar_filter_parameters["solver_settings"].AddDouble("filter_radius",filter_radius)
+    shell_scalar_filter_parameters["solver_settings"].AddString("model_part_name",model_part_name)
+    shell_scalar_filter_parameters["solver_settings"].AddValue("linear_solver_settings", linear_solver_settings)
+
+    fixed_model_parts = parameters["fixed_model_parts"].GetStringArray()
+
+    for fixed_model_part in fixed_model_parts:
+        auto_process_settings = Kratos.Parameters(
+            """
+            {
+                "python_module" : "fix_scalar_variable_process",
+                "kratos_module" : "KratosMultiphysics",
+                "process_name"  : "FixScalarVariableProcess",
+                "Parameters"    : {
+                    "model_part_name" : \""""+fixed_model_part+"""\",
+                    "variable_name"   : "HELMHOLTZ_SCALAR",
+                    "constrained"     : true
+                }
+            }
+            """)
+        shell_scalar_filter_parameters["processes"]["boundary_conditions_process_list"].Append(auto_process_settings)
+
+    return shell_scalar_filter_parameters
+
+class ShellThicknessControl(Control):
+    """Shell thickness control
+
+    This is filtering-based discrete thickness control which parametrizes and controls
+    shell elements thickness.
+
+    """
+    def __init__(self, name: str, model: Kratos.Model, parameters: Kratos.Parameters, optimization_problem: OptimizationProblem):
+        super().__init__(name)
+
+        self.parameters = parameters
+        self.model = model
+        self.optimization_problem = optimization_problem
+
+        default_settings = Kratos.Parameters("""{
+            "controlled_model_part_names": [],
+            "filter_settings": {
+                "type" : "implicit",
+                "radius": 0.000000000001,
+                "linear_solver_settings" : {}
+            },
+            "beta_settings": {
+                "initial_value" : 25,
+                "max_value" : 25,
+                "adaptive" : false,
+                "increase_fac" : 1.5,
+                "update_period" : 20
+            },
+            "SIMP_power_fac": 3,
+            "initial_thickness":0.000001,
+            "physical_thicknesses": [],
+            "fixed_model_parts": [],
+            "fixed_model_parts_thicknesses": [],
```

Comment:
I would encapsulate this to a dictionary where key is the model part name and value is the thickness. So the defaults will looklike the following:
```suggestion
            "fixed_model_part_and_thicknesses": {
                "MODEL_PART_NAME": 1.0
            },
```